src/components/ui/Button/index.js

```diff
@@ -58,6 +70,8 @@ export const Button = forwardRef((props, ref) => {
       ref={ref}
       {...otherProps}
     >
+      {leadingIcon}
+      {'\u00A0'}
```

Comment:
I appreciate you appeasing the linter, but I think it points to a deeper problem: this space is only being used to create a visual separation, so really the icon and content should be next to each other but separated with a margin/CSS. That should give you more fine-grained control over the separation and be more semantically accurate 😄

Reply:
You're not wrong, but it's still more annoying to do and I'm lazy. 

Fixed now! Test and/or critique away!